velox/common/memory/ByteStream.cpp

```diff
@@ -172,6 +172,31 @@ std::string_view BufferInputStream::nextView(int32_t size) {
       reinterpret_cast<char*>(current_->buffer) + position, viewSize);
 }
 
+std::unique_ptr<folly::IOBuf> BufferInputStream::readBytes(int32_t size) {
+  VELOX_CHECK_GE(size, 0, "Attempting to read negative number of bytes");
+  if (size == 0) {
+    return nullptr;
+  }
+  std::unique_ptr<folly::IOBuf> result;
+  for (;;) {
+    const int32_t availableBytes = current_->size - current_->position;
+    const int32_t readBytes = std::min(availableBytes, size);
+    auto newBuf = folly::IOBuf::wrapBuffer(
+        current_->buffer + current_->position, readBytes);
+    if (result) {
+      result->prev()->appendChain(std::move(newBuf));
```

Comment:
The order of buffers here is incorrect.  Just do `result->appendToChain(std::move(newBuf))`.

Reply:
The order is same with IOBufOutputStream::getIOBuf(), appendToChain is also OK. I will change as you say.